avocado/plugins/run.py

```diff
@@ -166,7 +168,18 @@ def run(self, args):
             log.error(e.message)
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
-        job_run = job_instance.run()
+
+        try:
+            # Run JobPre plugins
+            pre_post_dispatcher = JobPrePostDispatcher()
```

Comment:
When this line fails you're accessing non-existing variable. Please move this line outside the `try/finally` block.